AzureCommunicationUI/AzureCommunicationUI/Presentation/Style/LocalizationProvider.swift

```
@@ -21,9 +25,48 @@ struct LocalizationProvider {
         }
 
         LocalizationProvider.locale = localeConfig.locale
+        LocalizationProvider.localizableFilename = localeConfig.localizableFilename
+        LocalizationProvider.customTranslations = localeConfig.customTranslations
     }
 
     static func getSupportedLanguages() -> [String] {
         return supportedLocales
     }
+
+    static func getLocalizedString(_ key: String) -> String {
+        if let customTranslation = customTranslations[key] {
+            return customTranslation
+        }
+
+        if let path = Bundle.main
+            .path(forResource: locale, ofType: "lproj"),
+           let bundle = Bundle(path: path) {
+            let customLocalizableString = NSLocalizedString(key,
+                                                            tableName: localizableFilename,
+                                                            bundle: bundle,
+                                                            value: "localize_key_not_found",
+                                                            comment: key)
+            if customLocalizableString != "localize_key_not_found" {
+                return customLocalizableString
+            }
```

In the case of a key not found should we return some error in the `didFail` event that contoso can listen to? Cause right now we are just failing silently

---

If a key is not found, it's not an error, but part of the 4 step logic (Contoso did not supply custom translations).
1. Check if key exist in customTranslations dictionary
2. Check if key exist in Contoso provided Localizable.strings file for the locale
3. Check if specified valid locale, then return predefined translation string
4. return english as fallback

We will need to specify the default return value because how NSLocalizedString
* If key is not found and value is nil or an empty string, return key
* If key is not found and value is non-nil and not empty, return value